### PR TITLE
Added ENS governance subgraph

### PR DESCRIPTION
**Pull request by mmnavarr**
# Title
Added the ENS governance subgraph

## Description
This subgraph interacts with the `ENSGovernor` contract as well as the `ENSToken` contract to track proposals, delegates, and votes.

Indexing here -- https://thegraph.com/studio/subgraph/ens-governance/

codegen + build proof:
image

### CODE REVIEW

File: subgraphs/ens-governance/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  proposalsCanceled: BigInt!
+
+  "Quorum fraction numerator value. (quorum = totalSupply * numerator / denominator)"
+  quorumNumerator: BigInt
+  "Quorum fraction denominator value. (quorum = totalSupply * numerator / denominator)"
+  quorumDenominator: BigInt
+}
+
+type Proposal @entity {
+  id: ID!
```
Comment (this-username-is-taken) [inline subgraphs/ens-governance/schema.graphql:34]: Should document every field. e.g. what does the ID look like? The goal is for someone be able to figure out what each field is/does just by looking at the schema in the subgraph playground

File: subgraphs/ens-governance/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+enum VoteChoice {
+  FOR
+  AGAINST
+  ABSTAIN
+}
+
+type TokenHolder @entity {
+  "A TokenHolder is any address that holds any amount of Compound Tokens, the id used is the blockchain address."
+  id: ID!
```
Comment (this-username-is-taken) [inline subgraphs/ens-governance/schema.graphql:116]: Consider using `Bytes` for performance reasons: https://medium.com/edge-node-engineering/two-simple-subgraph-performance-improvements-a76c6b3e7eac

File: subgraphs/ens-governance/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  ACTIVE
+  CANCELED
+  DEFEATED
+  SUCCEEDED
+  QUEUED
+  EXPIRED
+  EXECUTED
+}
+
+type Vote @entity {
```
Comment (this-username-is-taken) [inline subgraphs/ens-governance/schema.graphql:93]: If this does get updated after creation, consider using `immutable`: https://medium.com/edge-node-engineering/two-simple-subgraph-performance-improvements-a76c6b3e7eac

File: subgraphs/ens-governance/src/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+} from "../generated/schema";
+
+export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const BIGINT_ZERO = BigInt.fromI32(0);
+export const BIGINT_ONE = BigInt.fromI32(1);
+export const BIGINT_FIVE = BigInt.fromI32(5);
+export const BIGDECIMAL_ZERO = new BigDecimal(BIGINT_ZERO);
+
+export const GOVERNANCE_NAME = "OZGovernor";
+export const PROPOSAL_STATE_PENDING = "PENDING";
```
Comment (this-username-is-taken) [inline subgraphs/ens-governance/src/helpers.ts:16]: I feel like something like this is cleaner:
``` typescript
export namespace ProposalState {
  export const PENDING = "PENDING";
  ...
}
```

**Comment by this-username-is-taken** [comment]
Minor suggestions but otherwise all looks good to me!

File: subgraphs/ens-governance/src/helpers.ts (hunk 4 above)
Comment (mmnavarr) [reply subgraphs/ens-governance/src/helpers.ts:18]: Agh this is much cleaner. I was using enums before but AssemblyScript wasn't very happy with me

**Comment by mmnavarr** [comment]
Converted all `String` addresses to `Bytes` per your recommendation 👍🏼

**Comment by this-username-is-taken** [comment]
LGTM!

**Pull request merged**